registry/golang/container.yaml

```diff
@@ -3,16 +3,17 @@ url: https://hub.docker.com/r/_/golang
 maintainer: '@vsoch'
 description: Go (a.k.a., Golang) is a programming language first developed at Google.
 latest:
-  1.18-rc: sha256:c9be2380bfdc0f4da95e1635e914dbd596bc73b618cdf16894a631eb58945156
+  1-cross: sha256:bb3578192ab4a6b744819de9811d0b494a7b8444490e4c6f13926aa201163925
 tags:
-  1.16.4-alpine: sha256:0dc62c5cc2d97657c17ff3bc0224214e10226e245c94317e352ee8a2c54368b4
-  1.16.5-alpine: sha256:45f32e963bb3cc408cfcd01a8e76b2872fb238f602ec5481cd75393da29369c0
+  1.18-rc: sha256:2da497bcc0c9ff09d4185907068c6f137d14e8848059971072f2e9cc936aae70
   1.17-rc-alpine: sha256:787111a3069abdb2c4d8c0b27dff2a29cef8b147f8e7a431f5a464ea84ebfa41
-  1.17.0: sha256:7dbfeb9d51c049e8bfe36cf1a4217c7b1ba304bf0eb72d57d0c04f405589f122
-  1.17.2: sha256:124966f5d54a41317ee81ccfe5f849d4f0deef4ed3c5c32c20be855c51c15027
-  1.17.3: sha256:199102125d11c943c927a8a33911ef960ca72c4879e307c7c2e40ceaa72201e3
-  1.18-rc: sha256:c9be2380bfdc0f4da95e1635e914dbd596bc73b618cdf16894a631eb58945156
-  latest: sha256:e06c83493ef6d69c95018da90f2887bf337470db074d3c648b8b648d8e3c441e
+  1.16.5-alpine: sha256:45f32e963bb3cc408cfcd01a8e76b2872fb238f602ec5481cd75393da29369c0
+  latest: sha256:5eb58ca0a747ed2e2f4e069d1116badb02a172cf160d31f801776a2342c12863
+  '1': sha256:5eb58ca0a747ed2e2f4e069d1116badb02a172cf160d31f801776a2342c12863
+  1-alpine: sha256:3afd220509acf9866e91932a3a41bf341b8bada82107ef3ecce3422826b98064
+  1-bullseye: sha256:da371518962826141d504029f0e8c27c64f57c69d2899ebf3c9a9e7afb136a10
+  1-buster: sha256:284da7e6ad7fd1250a785c24912fb6f8652176b588d637de5c5b3d4a4aa1693a
+  1-cross: sha256:bb3578192ab4a6b744819de9811d0b494a7b8444490e4c6f13926aa201163925
```

Comment:
That's an interesting one, and sort of the same issue as `openjdk`, `php`, `mysql`, `postgres`, etc. Since `shpc` focuses on reproducibility, ideally it should favour the full versions such as `1.18.0-buster` rather than `1-buster`

And since it only considers the latest _n_ tags, you could add a filter to only list say  the `*-buster` tags. I imagine that on an HPC someone would rather pin the underlying distribution, and get several versions on it, rather than essentially pinning the version and getting all the distro flavours. But I see that'd be a controversial change.

Reply:
The reason is because the way we do a comparison (and the way these StrictVersion and LooseVersions work) is by way of tuple comparison. So anything that is a list of numbers we can compare! For example using some of the above:

```
In [1]: [1] < [1,1,1]
Out[1]: True

In [2]: [1] < [1,1]
Out[2]: True

In [3]: [1] < [0,1]
Out[3]: False
```
And so we are technically reading all of those as just a version 1. Here is what I'll try for the next run - doing the same "don't allow duplicates" but not just for major minor, for just major too. So here there is just one version of a 1-<some string> container.

![image](https://user-images.githubusercontent.com/814322/161669716-d967e4ec-ea6f-4b6e-9521-952787fc44ec.png)

This is probably a good idea - I'll have this in for the next run (it sounds like this is a common issue, and akin to have a gazillion versions of 1.17 we should probably not have a gazillion of 1).